File: testdata/jennies/builders/composable_slot/PHPBuilder/src/ComposableSlot/LokiBuilderBuilder.php
<?php

namespace Grafana\Foundation\ComposableSlot;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\ComposableSlot\Dashboard>
 */
class LokiBuilderBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\ComposableSlot\Dashboard $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\ComposableSlot\Dashboard();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\ComposableSlot\Dashboard
     */
    public function build()
    {
        return $this->internal;
    }

    /**
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Cog\Dataquery> $target
     */
    public function target(\Grafana\Foundation\Cog\Builder $target): static
    {
        $targetResource = $target->build();
        $this->internal->target = $targetResource;
    
        return $this;
    }

    /**
     * @param array<\Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Cog\Dataquery>> $targets
     */
    public function targets(array $targets): static
    {
            $targetsResources = [];
            foreach ($targets as $r1) {
                    $targetsResources[] = $r1->build();
            }
        $this->internal->targets = $targetsResources;
    
        return $this;
    }

}

File: testdata/jennies/builders/composable_slot/CSharpBuilder/src/Grafana/Foundation/ComposableSlot/LokiBuilderBuilder.cs
namespace Grafana.Foundation.ComposableSlot;


public class LokiBuilderBuilder : Cog.IBuilder<Dashboard>
{
    protected readonly Dashboard @internal;

    public LokiBuilderBuilder()
    {
        this.@internal = new Dashboard();
    }

    public LokiBuilderBuilder Target(Cog.IBuilder<Cog.Variants.Dataquery> target)
    {
        Cog.Variants.Dataquery targetResource = target.Build();
        this.@internal.Target = targetResource;
        return this;
    }

    public LokiBuilderBuilder Targets(List<Cog.IBuilder<Cog.Variants.Dataquery>> targets)
    {
        List<Cog.Variants.Dataquery> targetsResources = new List<Cog.Variants.Dataquery>();
        foreach (Cog.IBuilder<Cog.Variants.Dataquery> r1 in targets)
        {
                Cog.Variants.Dataquery targetsDepth1 = r1.Build();
                targetsResources.Add(targetsDepth1);
        }
        this.@internal.Targets = targetsResources;
        return this;
    }

    public Dashboard Build()
    {
        return this.@internal;
    }
}
